### PR TITLE
Added stackoverflow embeddings notebook

### PR DESCRIPTION
**Pull request by ivanmkc**
**REQUIRED:** Add a summary of your PR here, typically including why the change is needed and what was changed. Include any design alternatives for discussion purposes.

Added notebook that demonstrates how to generate stackoverflow embeddings.

Fixes errors in other notebooks being tested.

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [x] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

File: notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb [hunk 1]
```diff
@@ -0,0 +1,1131 @@
+{
```
Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:1]: <p>Nit, add in the article in bold "These are uploaded to <strong>the</strong> Vertex AI Matching Engine service"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:1]: <p>In the first sentence there seems to be an extra space before 'Approximate Nearest Neighbor'</p><p> </p><p>Nit: Make sure you lowercase instances of "Index"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:1]: <p>Nit, add comma after BigQuery, and add "the" before Vertex. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:1]: <p>same comment about bold text in the second sentence </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:1]: <p>comment about the numbered steps being inconsistent with rest of notebook, suggestion to remove </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:1]: <p>Is the 1000 supposed to be there at the end of this section?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (sarahcdugan) [inline notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:1]: <p>Maybe spell out GCS to be safe</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1575/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='45'/>

Comment (ivanmkc) [reply notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:1]: <p>Done.</p>

Comment (ivanmkc) [reply notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:1]: <p>Fixed</p>

Comment (ivanmkc) [reply notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:1]: <p>You mean, unbold it?</p>

Comment (ivanmkc) [reply notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:1]: <p>Ah it is the standard format in the notebook template.</p>

Comment (ivanmkc) [reply notebooks/official/matching_engine/sdk_matching_engine_create_stack_overflow_embeddings.ipynb:1]: <p>Will keep to be consistent unless we change the template.</p>

**Review by sarahcdugan — APPROVED** [review]

**Review by andrewferlitsch — APPROVED** [review]

**Pull request merged**